### PR TITLE
Menu load refactor

### PR DESCRIPTION
**Pull request by wpride**
Move/unify the menu loading code into core from MenuAdapter and MenuScreen

Open for review but haven't tested yet

cross-request: https://github.com/dimagi/commcare-android/pull/1611

### CODE REVIEW

**Comment by ctsims** [comment]
@wpride "Open for review but haven't tested yet" <- Is this tested now and happy to be pulled?

**Comment by wpride** [comment]
@ctsims made one update tested and confirmed this is working, good to review/pull

File: src/main/java/org/commcare/suite/model/MenuLoader.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                      LoggerInterface loggerInterface) {
+        this.loggerInterface = loggerInterface;
+        this.getMenuDisplayables(platform, sessionWrapper, menuId);
+    }
+
+    public String getErrorMessage() {
+        if (loadException != null) {
+            String errorMessage = loadException.getMessage();
+            if (loadException instanceof XPathSyntaxException) {
+                errorMessage = Localization.get("app.menu.display.cond.bad.xpath", new String[]{xPathErrorMessage, loadException.getMessage()});
```
Comment (ctsims) [inline src/main/java/org/commcare/suite/model/MenuLoader.java]: Hey,

So currently there shouldn't be an assumption that there are any "universal" strings in the localization engine like this, so wee should likely move these to something like loggerInterface.logXPathParseError, loggerInterface.logXPathException() or moving the errors to be non-localized.

**Comment by ctsims** [comment]
@damagatchi retest this please

**Comment by ctsims** [comment]
@damagatchi retest this please

**Pull request merged**